deeppavlov/dataset_readers/insurance_reader.py

```diff
@@ -1,51 +1,54 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 from pathlib import Path
+from typing import Dict, List, Tuple
+
+from deeppavlov.core.data.dataset_reader import DatasetReader
 from deeppavlov.core.common.registry import register
-from deeppavlov.core.data.utils import download_decompress, mark_done, is_done
-from deeppavlov.core.commands.utils import get_deeppavlov_root, expand_path
-from typing import Dict, List, Union
+from deeppavlov.core.commands.utils import expand_path
 
 
 @register('insurance_reader')
 class InsuranceReader(DatasetReader):
-    
-    def read(self, data_path: str, **kwargs) -> Dict[str, List[Dict[str, Union[int, List[int]]]]]:
-        """Read the InsuranceQA data from files and forms the dataset.
+    """The class to read the InsuranceQA V1 dataset from files.
 
-        Args:
-            data_path: A path to a folder where dataset files are stored.
-            **kwargs: Other parameters.
+    Please, see https://github.com/shuzi/insuranceQA.
 
-        Returns:
-            A dictionary containing training, validation and test parts of the dataset obtainable via
-            ``train``, ``valid`` and ``test`` keys.
-        """
+    Args:
+        data_path: A path to a folder with dataset files.
+        num_samples: A number of data samples to use in ``train``, ``validation`` and ``test`` mode.
+    """
 
+    def read(self, data_path: str, num_samples: int = None, **kwargs) -> Dict[str, List[Tuple[List[str], int]]]:
         data_path = expand_path(data_path)
-        self._download_data(data_path)
         dataset = {'train': None, 'valid': None, 'test': None}
         train_fname = Path(data_path) / 'insuranceQA-master/V1/question.train.token_idx.label'
         valid_fname = Path(data_path) / 'insuranceQA-master/V1/question.dev.label.token_idx.pool'
         test_fname = Path(data_path) / 'insuranceQA-master/V1/question.test1.label.token_idx.pool'
-        self.idxs2cont_vocab = self._build_context2toks_vocabulary(train_fname, valid_fname, test_fname)
-        dataset["valid"] = self._preprocess_data_valid_test(valid_fname)
-        dataset["train"] = self._preprocess_data_train(train_fname)
-        dataset["test"] = self._preprocess_data_valid_test(test_fname)
+        int2tok_fname = Path(data_path) / 'insuranceQA-master/V1/vocabulary'
+        response2ints_fname = Path(data_path) / 'insuranceQA-master/V1/answers.label.token_idx'
```

Comment:
`data_path` is already a `Path` after the `expand_path` function

Reply:
corrected